### ISSUE TITLE
License

### ISSUE THREAD

**Issue by aperezhortal** [samhatfield/speedy.f90#7]
Hi @samhatfield , the repo does not have a license. Do you plan to add one?

**Comment by samhatfield** [comment]
Hi @aperezhortal, to be honest I'm not sure which license is most appropriate. I'd like to have the most open license possible, but I'd have to get the blessing of the original developers (Franco Molteni and Fred Kucharski) first as I don't believe they distributed SPEEDY with a license originally, at least I couldn't find one on the [website](http://users.ictp.it/~kucharsk/speedy-net.html).

Which license would you suggest?

**Comment by aperezhortal** [comment]
I have an older version of the SPEEDY code sent by Fred Kucharski and it does not have any license. So, it is a good idea to get in touch with the original developers.

Two popular licenses used in academia are [MIT](https://tldrlegal.com/license/mit-license) and [BSD 3 Clause](https://tldrlegal.com/license/bsd-3-clause-license-(revised)). Those are permissive licenses that allow doing almost anything, as long as you redistribute the code with the original copyright notice. 

Using permissive licenses can help to attract more collaborators to the project. 

By the way, thanks for the work you have done here!

**Comment by gaelforget** [comment]
Hi. 

I just sent PR with a possible license. Please see https://github.com/samhatfield/speedy.f90/pull/8 for a bit more detail

On a related note, I might suggest creating a release with a DOI / zenodo archive once you have a license. Anyone else can do that from their forks of course but it would be great if we could refer to original author / source for this translation into f90

https://guides.github.com is useful on these topics imho -- see [making your code citable](https://guides.github.com/activities/citable-code/) for example

**Comment by samhatfield** [comment]
Hi @gaelforget - thanks for the suggested license and for reminding me of this issue. I've just sent an email to the original developers , quoting your suggested license, so we will see what they say. This repo seems to be gaining some interest so it would be a shame to have to take it down due to licensing issues with the original code.

**Comment by gaelforget** [comment]
> Hi @gaelforget - thanks for the suggested license and for reminding me of this issue. I've just sent an email to the original developers , quoting your suggested license, so we will see what they say. 

Great! Thank you!!

> This repo seems to be gaining some interest so it would be a shame to have to take it down due to licensing issues with the original code.

Agreed. Seems like a no-brainer to me although opinions can differ.

For context, I am looking at it as a candidate for the initial suite of model examples for [ClimateModels.jl](https://github.com/gaelforget/ClimateModels.jl). I have a working integration that's good enough for now but consolidating this repo (license + zenodo) seems needed to ensure that users can run reliably and freely run the model via cloud service like [mybinder.org](https://mybinder.org) in the future (unlike with e.g. an _email me_ type gate). 

As an alternative, I could just use the SPEEDY version (v23) that we currently have in [MITgcm](https://mitgcm.readthedocs.io/en/latest/phys_pkgs/aim.html?highlight=speedy) but that would seem less than ideal to me.

**Comment by samhatfield** [comment]
Very nice - I hadn't come across ClimateModels.jl before! How did you deal with the lack of SPEEDY license for aim_v23? Or is that similarly in a gray area?

**Comment by gaelforget** [comment]
> Very nice - I hadn't come across ClimateModels.jl before! How did you deal with the lack of SPEEDY license for aim_v23? Or is that similarly in a gray area?

I wasn't really involved in the [coupling / port of SPEEDY to MITgcm](https://github.com/MITgcm/MITgcm/commit/d676f916b2a313d8369d3dff99d2e25f2c6ffef5) so I am not sure. However, I may note that this goes back to a very different era when things like this were'nt so well charted / understood (2002!). In fact it is only way later, in 2016, that we resolved a similar issue with MITgcm by adding [LICENSE.txt](https://github.com/MITgcm/MITgcm/commit/2567525c0295bce6b6143d0daaf5521dab03ac21#diff-d0ed4cc3fb70489fe51c7e0ac180cba2a7472124f9f9e9ae67b01a37fbd580b7) in that repo.

**Comment by samhatfield** [comment]
I see! Well I'm happy to say that Franco Molteni got back to me, and simply asked for commercial use to be removed, and mention of the original SPEEDY to be added. This was his preferred license, based on your suggested license:

```
Copyright (c) 2020 <Fred Kucharski, Franco Molteni, Sam Hatfield>

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software for research, educational and other non-commercial purposes, 
including the rights to use, copy, modify, merge, publish, distribute
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE 
FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, 
TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE 
OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.

Please note that the software is different from the version of the SPEEDY model distributed 
by the Abdus Salam International Centre for Theoretical Physics.
```
So I will add this file myself and close your PR. Thanks for bringing this up, and I hope that license is good enough for your purposes.

**Comment by samhatfield** [comment]
Closed by 7979838da2e9874ed51cbed217b77a6a4bfc1d7e.

**Comment by gaelforget** [comment]
> I see! Well I'm happy to say that Franco Molteni got back to me, and simply asked for commercial use to be removed, and mention of the original SPEEDY to be added. This was his preferred license, based on your suggested license:
> 
> ```
> Copyright (c) 2020 <Fred Kucharski, Franco Molteni, Sam Hatfield>
> 
> Permission is hereby granted, free of charge, to any person obtaining a copy
> of this software and associated documentation files (the "Software"), to deal
> in the Software for research, educational and other non-commercial purposes, 
> including the rights to use, copy, modify, merge, publish, distribute
> copies of the Software, and to permit persons to whom the Software is
> furnished to do so, subject to the following conditions:
> 
> The above copyright notice and this permission notice shall be included in all
> copies or substantial portions of the Software.
> 
> THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
> IMPLIED. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE 
> FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, 
> TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE 
> OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
> 
> Please note that the software is different from the version of the SPEEDY model distributed 
> by the Abdus Salam International Centre for Theoretical Physics.
> ```
> 
> So I will add this file myself and close your PR. Thanks for bringing this up, and I hope that license is good enough for your purposes.

Great! Thanks for resolving this
